sorc/regrid_nemsio.fd/fv3_interface.f90

```diff
@@ -62,7 +62,7 @@ subroutine fv3_regrid_nemsio()
     implicit none
     type(analysis_grid)                                                  :: anlygrd(ngrids)
     type(varinfo), allocatable, dimension(:)                             :: var_info,var_info2d,var_info3d
-    type(gfs_grid)                                                       :: gfs_grid
+    type(gfs_grid)                                                       :: mgfs_grid
```

Comment:
I do not understand the point of the changes in this file. Why are you just renaming gfs_grid to mgfs_grid?

Reply:
I didn't want to make that change but apparently it is needed to get the code compiling with GNU compilers. Intel allows such declarations
`type(gfs_grid)                                                       :: gfs_grid`
in which the type name and variable name are the same. GNU gfortran OTOH complains about it so I was forced to change the variable name. Maybe there better solutions than this  but I didn't invest much time on it finding out.

Reply:
Aha! I didn't notice the type and variable were the same name. Makes sense.

Reply:
I wonder if this is resolved in the develop branch of the ufs-weather-model repo. I would think it would have to be since they do have cmake that supports gcc.

Reply:
Oh, nevermind. this is from a different place.